Source/WarpX.cpp

```diff
@@ -1558,18 +1562,20 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
         const int rho_ncomps = (WarpX::do_multi_J) ? ncomps : 2*ncomps;
         rho_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba,rho_nodal_flag),dm,rho_ncomps,ngRho,tag("rho_fp"));
 
-        // Also allocate the MultiFab for the full rho grid
-        // get the full domain
-        Box domain_box = Geom(lev).Domain();
-        // make a new BoxArray from the domain Box
-        BoxArray ba_full(domain_box);
-        // make a DistributionMapping from the new BoxArray
-        DistributionMapping dm_full;
-        // force that distribution mapping to only go to the root proc
-        Vector<int> pmap = {0};
-        dm_full.define(pmap);
-        // make a FabArray to hold the phi data
-        full_rho_fp[lev] = std::make_unique<MultiFab>(amrex::convert(ba_full,rho_nodal_flag),dm_full,2*ncomps,ngRho,tag("full_rho_fp"));
+        if (!external_solver) {
```

Comment:
This should be if an external solver is used, so
```suggestion
        if (external_solver) {
```
right?

Reply:
Right, I misread the card..